internal/transformers/commonjsmodule.go

```diff
@@ -63,6 +63,8 @@ func (tx *CommonJSModuleTransformer) visitTopLevel(node *ast.Node) *ast.Node {
 	switch node.Kind {
 	case ast.KindImportDeclaration:
 		node = tx.visitTopLevelImportDeclaration(node.AsImportDeclaration())
+	case ast.KindJSImportDeclaration:
```

Comment:
IMO, these should be handled by something like `typeeraser.go` or `importellision.go` so that other transformers shouldn't have to deal with special cases like `JSImportDeclaration`. Handling them in the module transforms means they have to exist as part of all other transforms first, and its better if we limit syntax in those transforms to ones expected by that edition of JS.